internal_comps/gp_board/src/services/move-validator.js

```diff
@@ -9,23 +10,142 @@ import { GAME_KEYWORDS } from "../../../gp_game/src/entities/game-keywords";
   pieceY: the y (row) coordinate of the piece.
   board: the state of the board.  Of the following format:
   [
-      [
-          {
-              "team": GAME_KEYWORDS.BLACK_TEAM, // which team is on this space
-              "playerPiece": GAME_KEYWORDS.PAWN, // undefined if the player does not have a piece here, otherwise, one of the values from GAME_PIECE. 
-              <GAME_KEYWORDS.PAWN>: 0, // where <GAME_KEYWORDS.PAWN> is actually "pawn", but you should always use the reference like so: `board[0][0][GAME_KEYWORDS.PAWN]` to ensure consistency across the product.
-              <GAME_KEYWORDS.ROOK>: 50, // number of this type of piece on that cell. Only matters if it's 0 or > 0.
-              <GAME_KEYWORDS.KNIGHT>: 0,
-              <GAME_KEYWORDS.BISHOP>: 0,
-              <GAME_KEYWORDS.QUEEN>: 0,
-              <GAME_KEYWORDS.KING>: 0
-          },
-          ...
-      ],
+    [
+      {
+        "team": GAME_KEYWORDS.BLACK_TEAM, // which team is on this space
+        "playerPiece": GAME_KEYWORDS.PAWN, // undefined if the player does not have a piece here, otherwise, one of the values from GAME_PIECE. 
+        <GAME_KEYWORDS.PAWN>: 0, // where <GAME_KEYWORDS.PAWN> is actually "pawn", but you should always use the reference like so: `board[0][0][GAME_KEYWORDS.PAWN]` to ensure consistency across the product.
+        <GAME_KEYWORDS.ROOK>: 50, // number of this type of piece on that cell. Only matters if it's 0 or > 0.
+        <GAME_KEYWORDS.KNIGHT>: 0,
+        <GAME_KEYWORDS.BISHOP>: 0,
+        <GAME_KEYWORDS.QUEEN>: 0,
+        <GAME_KEYWORDS.KING>: 0
+      },
       ...
+    ],
+    ...
   ]
 */
 export function getValidMoves(pieceTeam, pieceType, pieceX, pieceY, board) {
-  debugger;
-  return [{ x: 0, y: 0 }, { x: 1, y: 0 }, { x: 4, y: 4 }, { x: 3, y: 4 }, { x: 3, y: 4 }, { x: 7, y: 7 }, { x: 6, y: 7 }];
+  return _getMoves(pieceTeam, pieceType, pieceX, pieceY, board);
 }
+
+// @TODO: ADD stop when hits piece
+function _getMoves(pieceTeam, pieceType, pieceX, pieceY, board) {
+  switch(pieceType) {
+    case GAME_KEYWORDS.PAWN:
+      return _pawnMoves(pieceTeam, pieceX, pieceY, board);
+    case GAME_KEYWORDS.ROOK:
+      return _rookMoves(pieceTeam, pieceX, pieceY, board);
```

Comment:
I originally didn't have that in there, but I thought I might need it when checking for piece collisions, but I may be able to get the info out of board so I will delete them if I don't use them.  The original code I tried to commit didn't have those because they weren't needed for what I have done so far.

Reply:
You can add them back in when needed.  I don't like stale variables...